test/octree/test_octree_iterator.cpp

```diff
@@ -810,6 +1016,418 @@ TEST_F (OctreePointCloudAdjacencyBeginEndIteratorsTest, BreadthEnd)
+
+    // Setup the octree
+    double resolution = 1.0 / (pow (2, depth_));
+    oct_.setResolution (resolution);
+
+    // Add points to octree
+    oct_.setInputCloud (cloud);
+
+    // Bounding box
+    oct_.defineBoundingBox (0.0, 0.0, 0.0, 1.0, 1.0, 1.0);
+
+    // Add points from cloud to octree
+    oct_.addPointsFromInputCloud ();
+  }
+
+  // Generate a vector of Sierpinski voxels at a given 'depth_arg'
+  vector<pair<Eigen::Vector3f, Eigen::Vector3f> >
+  generateSierpinskiVoxelExtremities (const Eigen::Vector3f & v_min, const Eigen::Vector3f & v_max,
+                                      const unsigned int & depth_arg)
+  {
+    pair<Eigen::Vector3f, Eigen::Vector3f> voxel (v_min, v_max);
+    vector<pair<Eigen::Vector3f, Eigen::Vector3f> > voxels;
+    voxels.push_back (voxel);
+
+    for (unsigned int i = 0; i < depth_arg; ++i)
+    {
+      voxels = generateSierpinskiVoxelExtremitiesAux (voxels);
+    }
+
+    return voxels;
+  }
+
+  // Apply a recursion step to a vector of macro Sierpinski voxel
+  // For each voxel in the input vector 'voxels_in', 4 sub-voxels are generated
+  vector<pair<Eigen::Vector3f, Eigen::Vector3f> >
+  generateSierpinskiVoxelExtremitiesAux (const vector<pair<Eigen::Vector3f, Eigen::Vector3f> > & voxels_in)
+  {
+    vector<pair<Eigen::Vector3f, Eigen::Vector3f> > voxels_out;
+
+    for (vector<pair<Eigen::Vector3f, Eigen::Vector3f> >::const_iterator it = voxels_in.begin ();
+        it != voxels_in.end ();
+        ++it)
+      {
+        Eigen::Vector3f v_min = it->first;
+        Eigen::Vector3f v_max = it->second;
+        Eigen::Vector3f v_mid = 0.5 * (v_min + v_max);
+
+        pair<Eigen::Vector3f, Eigen::Vector3f> voxel_0;
+        pair<Eigen::Vector3f, Eigen::Vector3f> voxel_1;
+        pair<Eigen::Vector3f, Eigen::Vector3f> voxel_2;
+        pair<Eigen::Vector3f, Eigen::Vector3f> voxel_3;
+
+        voxel_0.first  = v_min;
+        voxel_0.second = v_mid;
+        voxel_1.first  = Eigen::Vector3f (v_min.x (), v_mid.y (), v_mid.z ());
+        voxel_1.second = Eigen::Vector3f (v_mid.x (), v_max.y (), v_max.z ());
+        voxel_2.first  = Eigen::Vector3f (v_mid.x (), v_min.y (), v_mid.z ());
+        voxel_2.second = Eigen::Vector3f (v_max.x (), v_mid.y (), v_max.z ());
+        voxel_3.first  = Eigen::Vector3f (v_mid.x (), v_mid.y (), v_min.z ());
+        voxel_3.second = Eigen::Vector3f (v_max.x (), v_max.y (), v_mid.z ());
+
+        voxels_out.push_back (voxel_0);
+        voxels_out.push_back (voxel_1);
+        voxels_out.push_back (voxel_2);
+        voxels_out.push_back (voxel_3);
+      }
+
+    return voxels_out;
+  }
+
+  // Members
+  OctreeT oct_;
+
+  const unsigned depth_;
+};
+
+TEST_F (OctreePointCloudSierpinskiTest, DefaultIterator)
+{
+  // Useful types
+  typedef typename OctreeT::Iterator IteratorT;
+
+  // Check the number of branch and leaf nodes
+  ASSERT_EQ (oct_.getLeafCount (), pow (4, depth_));
+  ASSERT_EQ (oct_.getBranchCount (), (pow (4, depth_) - 1) / (4 - 1));
```

Comment:
Can you add a docstring with this url https://en.wikipedia.org/wiki/Geometric_progression#Geometric_series

Reply:
Since you use the expression so often you might want to add it as a static function to your `OctreePointCloudSierpinskiTest` class.

Reply:
So let me get this straight: You're able to use `oct_` at this scope, but invoking a static function which is kinda like a regular function from a practical standpoint, with `depth_` is not working?

That would trigger my OCD badly, but in this case is not important.

Reply:
I just try to do it. Logically, it should work, but it doesn't. I create a static method in the class `OctreePointCloudSierpinskiTest`:
```cpp
  static unsigned int 
  computeBranchCount ()
  {
    return (unsigned int)(pow (4, depth_) - 1) / (4 - 1);
  }
```

As it use the `depth_` field, so `depth_` should be declare as a static field. I do it:
```cpp
  const static unsigned depth_ = 7;
```
The `depth_` value never change, so hardcoded it's value. When I compile the code I get : 
```
[...]
Scanning dependencies of target test_octree_iterator
[100%] Building CXX object test/octree/CMakeFiles/test_octree_iterator.dir/test_octree_iterator.cpp.o
[100%] Linking CXX executable test_octree_iterator
CMakeFiles/test_octree_iterator.dir/test_octree_iterator.cpp.o : Dans la fonction « OctreePointCloudSierpinskiTest::SetUp() » :
/home/frozar/wk/pcl_repo/test/octree/test_octree_iterator.cpp:1090 : référence indéfinie vers « OctreePointCloudSierpinskiTest::depth_ »
collect2: error: ld returned 1 exit status
test/octree/CMakeFiles/test_octree_iterator.dir/build.make:106 : la recette pour la cible « test/octree/test_octree_iterator » a échouée
make[3]: *** [test/octree/test_octree_iterator] Erreur 1
CMakeFiles/Makefile2:5958 : la recette pour la cible « test/octree/CMakeFiles/test_octree_iterator.dir/all » a échouée
make[2]: *** [test/octree/CMakeFiles/test_octree_iterator.dir/all] Erreur 2
CMakeFiles/Makefile2:5970 : la recette pour la cible « test/octree/CMakeFiles/test_octree_iterator.dir/rule » a échouée
make[1]: *** [test/octree/CMakeFiles/test_octree_iterator.dir/rule] Erreur 2
Makefile:1703 : la recette pour la cible « test_octree_iterator » a échouée
make: *** [test_octree_iterator] Erreur 2
```

It's in french but the error is "undefined reference to « OctreePointCloudSierpinskiTest::depth_ »".

Reply:
I was bored :x https://github.com/SergioRAgostinho/pcl/commit/ba8b7ff3a9127e9a85e9c6187de6dab72b4b0a95 . It is indeed different from what you described, but this is what I originally had in mind when I made the comment. 

Cherry pick it/ squash or not. It's up to you.